src/lib/utils/hooks/useGetInfoFromIds.ts

```diff
@@ -80,46 +76,50 @@ export default function useGetInfoFromIds<T extends EntityHeader | UserProfile>(
 ) {
   const { token, ids, type } = props
   const [data, setData] = useState<Array<T>>([])
+  const [isLoading, setIsLoading] = useState(false)
 
   const idProp = (type: HookType) =>
     type === 'USER_PROFILE' ? 'ownerId' : 'id'
 
-  useEffect(() => {
+  const getData = useCallback(async () => {
+    if (isLoading) {
+      // We use isLoading to ensure that we don't load extra data while its process the data in batches
+      return
+    }
+    // look at current list of data, see if incoming ids has new data,
+    // if so grab those ids
+    const curList = data.map(el => el[idProp(type)])
+    const incomingList = ids.filter(el => el !== SynapseConstants.VALUE_NOT_SET)
+    const newValues = uniq(without(incomingList, ...curList))
+    if (newValues.length > 0) {
+      setIsLoading(true)
+      try {
+        const newIds = Array.from<string>(newValues)
+        const newReferences: LookupRequestType[] =
+          type === 'USER_PROFILE'
+            ? newIds
+            : newIds.map(el => ({ targetId: el }))
+        const newReferencesChunks = chunk(newReferences, 45)
 
-    const getData = async () => {
-      // look at current list of data, see if incoming ids has new data,
-      // if so grab those ids
-      const curList = data.map(el => el[idProp(type)])
-      const incomingList = ids.filter(
-        el => el !== SynapseConstants.VALUE_NOT_SET,
-      )
-      const newValues = without(incomingList, ...curList)
-      if (newValues.length > 0) {
-        try {
-          const newIds = Array.from<string>(newValues)
-          const newReferences: LookupRequestType[] =
+        for (const newReferences of newReferencesChunks) {
+          const newData =
             type === 'USER_PROFILE'
-              ? newIds
-              : newIds.map(el => ({ targetId: el }))
-          const newReferencesChunks = chunk(newReferences, 45)
-
-          for (const newReferences of newReferencesChunks) {
-            const newData =
-              type === 'USER_PROFILE'
-                ? await getUserProfileItems(newReferences as string[], token)
-                : await getEntityHeaderItems(
-                    newReferences as ReferenceList,
-                    token,
-                  )
-
-            setData(oldData => oldData.concat(...(newData as T[])))
-          }
-        } catch (error) {
-          console.error('Error on data retrieval', error)
+              ? await getUserProfileItems(newReferences as string[], token)
+              : await getEntityHeaderItems(
+                  newReferences as ReferenceList,
+                  token,
+                )
+          setData(oldData => oldData.concat(...(newData as T[])))
         }
+      } catch (error) {
+        console.error('Error on data retrieval', error)
+      } finally {
+        setIsLoading(false)
       }
     }
```

Comment:
there's an issue in all of our hooks components which store data over time - they'll hold stale data if the user ever log's in. If we use another useEffect that only takes in token as a dev dependency then we can capture if the token changes and retrieve the right data.